runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -1057,12 +1087,63 @@ else if (!isCorsRequestAllowed(server.binding, headers))
                             if (credentialsMatch != null)
                             {
                                 exchangeAuth = guard.reauthorize(server.initialId, credentialsMatch);
+                                if (exchangeAuth == 0)
+                                {
+                                    HttpEventFW event = server.httpEventRW
+                                        .wrap(server.eventBuffer, 0, server.eventBuffer.capacity())
+                                        .guardAuthorizationFailure(e -> e
+                                            .originId(server.originId)
+                                            .routedId(server.routedId)
+                                            .initialId(server.initialId)
+                                            .replyId(server.replyId)
+                                            .affinity(server.affinity)
+                                            .authorization(authorization)
+                                            .budgetId(budgetId)
+                                            .traceId(traceId)
+                                        )
+                                        .build();
+                                    System.out.println(event); // TODO: Ati
+                                }
+                                else
+                                {
+                                    HttpEventFW event = server.httpEventRW
+                                        .wrap(server.eventBuffer, 0, server.eventBuffer.capacity())
+                                        .guardAuthorizationSuccess(e -> e
+                                            .originId(server.originId)
+                                            .routedId(server.routedId)
+                                            .initialId(server.initialId)
+                                            .replyId(server.replyId)
+                                            .affinity(server.affinity)
+                                            .authorization(authorization)
+                                            .budgetId(budgetId)
+                                            .traceId(traceId)
+                                        )
+                                        .build();
+                                    System.out.println(event); // TODO: Ati
+                                }
```

Comment:
Perhaps this code can simplify if we use the same structure for success and failure, just a different event type indicator.

Reply:
yep, introduced
```
        enum Result
        {
            SUCCESS,
            FAILURE
        }
```
simplified the code indeed, and probably an abstraction that will be useful for the other event types, too